models_test.go

```diff
@@ -56,10 +56,16 @@ type Post struct {
 }
 
 type Comment struct {
-	ID       int    `jsonapi:"primary,comments"`
-	ClientID string `jsonapi:"client-id"`
-	PostID   int    `jsonapi:"attr,post_id"`
-	Body     string `jsonapi:"attr,body"`
+	ID       int                 `jsonapi:"primary,comments"`
+	ClientID string              `jsonapi:"client-id"`
+	PostID   int                 `jsonapi:"attr,post_id"`
+	Body     string              `jsonapi:"attr,body"`
+	Links    *Links              `jsonapi:"links,omitempty"`
+	Relation *SingleRelationLink `jsonapi:"relation,single-relation-link"`
```

Comment:
This is added because `relations` usually have a full set `data`  of data attributes, but according the [spec](https://jsonapi.org/format/#document-resource-object-relationships):

```
A  “relationship object” MUST contain at least one of the following:
* links: ...
* data: ...
* meta: ..
```

Right now, it only works for `data`, so this adds it for `links`.  _Note: I have not tested or checked to see if this works for meta, but my says it does not._